pkg/mimir/grpc_push_check.go

```diff
@@ -18,8 +18,8 @@ import (
 )
 
 type ingesterPushReceiver interface {
-	StartPushRequest(requestSize int64) error
-	FinishPushRequest(requestSize int64)
+	StartPushRequest(ctx context.Context, requestSize int64) (context.Context, error)
+	FinishPushRequest(ctx context.Context)
 }
```

Comment:
I think a nice side effect of these changes, is that we don't need two interfaces `ingesterPushReceiver` and `distributorPushReceiver` any more. Here I kept them in place, but want to refactor this chunk further later.